dwallet/src/eth_state.rs

```diff
@@ -0,0 +1,628 @@
+use std::cmp;
+use chrono::{Duration};
+use std::time::{SystemTime, UNIX_EPOCH};
+use eyre::{anyhow, Error, eyre};
+use milagro_bls::PublicKey;
+use ssz_rs::{Merkleized, Vector};
+use config::Network;
+use consensus::constants::MAX_REQUEST_LIGHT_CLIENT_UPDATES;
+use consensus::errors::ConsensusError;
+use consensus::{get_bits, get_participating_keys, is_current_committee_proof_valid, is_finality_proof_valid, is_next_committee_proof_valid};
+use consensus::rpc::ConsensusRpc;
+use consensus::rpc::nimbus_rpc::NimbusRpc;
+use consensus::types::{BLSPubKey, Bootstrap, Bytes32, FinalityUpdate, GenericUpdate, Header, OptimisticUpdate, SignatureBytes, SyncCommittee, Update, UpdatesResponse};
+use consensus::types::primitives::U64;
+use consensus::utils::{calc_sync_period, compute_domain, compute_signing_root, is_aggregate_valid};
+use tracing::info;
+
+/* NOTE:
+    This struct is created to have only the necessary functions for the proof verification.
+    Some functions in this struct are borrowed from the consensus module in the Helios project.
+    We had to copy them here since their logic is similar, but the types they operate on are different.
+*/
```

Comment:
`/**/` comments are unconventional in Rust.
Please turn this into a module-level documentation

Reply:
Please link the code you copied from.

Reply:
Please turn this into a module-level documentation